Week6/news/src/app/auth-guard.service.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from "@angular/core";
+import { Router, CanActivate } from "@angular/router";
+import { AuthService } from "./auth.service";
+@Injectable({
+  providedIn: "root"
+})
+export class AuthGuardService implements CanActivate {
+  constructor(private _authService: AuthService, private router: Router) {}
+  canActivate(): boolean {
+    if (!this._authService.isUserLoggedIn()) {
+      this.router.navigate([""]);
```

Comment:
navigate to login page

Reply:
The route("") itself contains login form , which was hidden intially . No seperate component for login page .